### PR TITLE
OPUS improvements

### PR DESCRIPTION
**Pull request by drnikolaev**
Some fixes and tunings

### CODE REVIEW

File: riva/clients/tts/riva_tts_client.cc [hunk 1]
```diff
@@ -124,7 +124,11 @@ main(int argc, char** argv)
 
   int32_t rate = FLAGS_rate;
   if (FLAGS_audio_encoding == "opus") {
-    rate = riva::utils::opus::Encoder::AdjustRateIfUnsupported(FLAGS_rate);
+    if (rate == 44100) {
+      rate = 48000;
+    } else {
+      rate = riva::utils::opus::Encoder::AdjustRateIfUnsupported(FLAGS_rate);
+    }
```
Comment (blisc) [inline riva/clients/tts/riva_tts_client.cc]: Let's revert this

File: riva/clients/tts/riva_tts_client.cc [hunk 2]
```diff
@@ -176,8 +175,7 @@ main(int argc, char** argv)
         std::chrono::duration<double> elapsed_first_audio = t_first_audio - start;
         std::cerr << "Time to first chunk: " << elapsed_first_audio.count() << " s" << std::endl;
       }
-      std::cerr << "Got chunk: " << chunk.audio().size() << " bytes" << std::endl;
-
+      std::cout << "Got chunk: " << chunk.audio().size() << " bytes" << std::endl;
```
Comment (blisc) [inline riva/clients/tts/riva_tts_client.cc]: Why aren't we using glog here?

File: riva/utils/opus/opus_client_decoder.cc [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  }
+  return ret;
+}
+
+int32_t
+Decoder::AdjustRateIfUnsupported(int32_t rate)
+{
+  int32_t adjusted_rate = 0;
+  if (rate < 800) {
+    adjusted_rate = 800;
```
Comment (blisc) [inline riva/utils/opus/opus_client_decoder.cc]: Shouldn't this be 8k?

File: riva/utils/opus/opus_client_decoder.cc [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+std::vector<int16_t>
+Decoder::DecodePcm(const std::vector<unsigned char>& packet)
+{
+  if (decoder_ == nullptr) {
+    int err;
+    decoder_ = opus_decoder_create(rate_, channels_, &err);
+    if (err < 0) {
+      LOG(ERROR) << "Failed to create encoder: " << opus_strerror(err);
```
Comment (blisc) [inline riva/utils/opus/opus_client_decoder.cc]:
```suggestion
      LOG(ERROR) << "Failed to create decoder: " << opus_strerror(err);
```

File: riva/clients/tts/riva_tts_client.cc (hunk 2 above)
Comment (drnikolaev) [reply riva/clients/tts/riva_tts_client.cc]: Idk, it's everywhere and I just follow the pattern

File: riva/utils/opus/opus_client_decoder.cc (hunk 3 above)
Comment (drnikolaev) [reply riva/utils/opus/opus_client_decoder.cc]: fixed, thnx

File: riva/clients/tts/riva_tts_client.cc (hunk 2 above)
Comment (blisc) [reply riva/clients/tts/riva_tts_client.cc]: Can you change it to glog please?

File: riva/utils/opus/opus_client_decoder.h [hunk 5]
```diff
@@ -74,34 +64,58 @@ class Decoder {
...
     return val;
   }
 
+  /**
+ * If requested rate is not supported this helper computes nearest supported one.
+ * Returns 0 if no adjustment required.
+ * @param rate
+ * @return
+ */
+  static int32_t AdjustRateIfUnsupported(int32_t rate);
```
Comment (blisc) [inline riva/utils/opus/opus_client_decoder.h]: Does this still return 0 if if no adjustment required?

File: riva/clients/tts/riva_tts_client.cc (hunk 2 above)
Comment (drnikolaev) [reply riva/clients/tts/riva_tts_client.cc]: done

File: riva/utils/opus/opus_client_decoder.h (hunk 5 above)
Comment (drnikolaev) [reply riva/utils/opus/opus_client_decoder.h]: No, it doesn't

**Review by blisc — APPROVED** [review]

**Pull request merged**